docs/tests/domdiff.js

```diff
@@ -0,0 +1,119 @@
+// eslint-env node, mocha
+
+const {Builder} = require('selenium-webdriver');
+const firefox = require('selenium-webdriver/firefox');
+const chrome = require('selenium-webdriver/chrome');
+const {join} = require('path');
+const mkdirp = require('mkdirp');
+const serve = require('serve');
+const pretty = require('pretty');
+const {writeFileSync, readFileSync, unlinkSync} = require('fs');
+const {expect} = require('chai');
+require('geckodriver');
+require('chromedriver');
+
+const goldenDir = 'golden';
+const testDir = 'generated';
+
+describe('docs site dom diff', async function() {
+  this.timeout(30000);
+
+  let server,
+    drivers = [];
+
+  before(async function() {
+    // noinspection JSPotentiallyInvalidUsageOfThis
+    this.timeout(15000);
+
+    const serveDir = join(__dirname, '../build');
+    server = serve(serveDir, {
+      port: 8080,
+      ignore: ['node_modules']
+    });
+
+    const ffoptions = new firefox.Options().headless();
+    const choptions = new chrome.Options().headless();
+
+    drivers.push({
+      driver: new Builder()
+        .forBrowser('firefox')
+        .setFirefoxOptions(ffoptions)
+        .build(),
+      name: 'firefox'
+    });
+    drivers.push({
+      driver: new Builder()
+        .forBrowser('chrome')
+        .setChromeOptions(choptions)
+        .build(),
+      name: 'chrome'
+    });
+
+    // And its wide screen/small screen subdirectories.
+    for (const dd of drivers) {
+      mkdirp.sync(join(__dirname, testDir, 'dom', dd.name, 'component'));
+    }
+
+    // Artificial wait as serve takes time to boot sometimes
+    await new Promise(resolve => {
+      setTimeout(() => resolve(), 2000);
+    });
+  });
+
+  after(() => {
+    server.stop();
+    for (const dd of drivers) {
+      dd.driver.quit();
+    }
+  });
+
+  this.slow(20000);
+
+  it("should match the Card page's dom against the golden directory", function() {
+    return compare_doms(drivers, 'component/card');
+  });
+});
+
+async function compare_doms(drivers, page) {
+  const generated = [];
+  for (const driver_desc of drivers) {
+    console.log(`Testing ${page} using ${driver_desc.name}`);
+    const {driver, name} = driver_desc;
+    await driver.get(`http://localhost:8080/${page}`);
+    await driver.sleep(2000);
+    const gen_dom = pretty(await driver.getPageSource(), {
+      ocd: true
+    })
+      .replace(
+        new RegExp('src="/(bundle|polyfills)([.]|-)[.a-zA-Z0-9]+[.]js"'),
```

Comment:
Can someone tell me what's wrong with these regexp's? They work in the Firefox console.

Reply:
You can use https://regex101.com/ to test your regex. It seems like the `/` after `src="` is not escaped.

Reply:
Using `new RegExp("[...]")` instead of `/[...]/`

Reply:
but yes will check that out